docs/source/tutorial_worlds.md

```diff
@@ -13,120 +13,353 @@ section. If you are interested in creating a task, please see
-
-![image](_static/img/world_batchbasic.png)
-
-There's a few more complex steps to actually completing a parley in this
-world.
-
-1.  Call `parley_init` on each shared world, if the world has it
-    implemented. Most classes don't need this--we currently only use it
-    for our `MultiWorld`, which handles the case when one specifies
-    multiple separate tasks to launch (e.g. `-t babi,squad`). This does
-    any pre-parley setup, here choosing which sub-tasks to use in the
-    next parley.
-2.  Then, iterate over the number of agents involved in the task. For
-    most tasks, this is just two agents: the teacher (task) and the
-    student (model). For each agent, we do two steps:
-
-    a.  Call `BatchWorld.batch_act()`. This method first checks if the
-        __original__ instance of the agent (not the copies) has a
-        function named `batch_act` implemented and does not have an
-        attribute `use_batch_act` set to `False`. This function is
-        described more below. If condition is not met, the BatchWorld's
-        `batch_act` method iterates through each agent copy in the batch
-        and calls the `act()` method for that instance. This is the
-        default behavior in most circumstances, and allows agents to
-        immediately work for batching without any extra work--the
-        batch\_act method is merely for improved efficiency.
-    b.  Call `BatchWorld.batch_observe()`. This method goes through
-        every __other__ agent, and tries to call the `observe()` method
-        on those agents. This gives other agents (usually, just the one
-        other agent) the chance to see the action of the agent whose
-        turn it is to act currently.
-
-Next, we'll look at how teachers and models can take advantage of the
-setup above to improve performance.
-
-### Batched Models
-
-Finally, models need to be able to handle observations arriving in
-batches.
-
-The first key concept to remember is that, if the model agent implements
-`batch_act()`, __act will not be called__ as long as `batchsize` > 1.
-
-However, copies of the agent will still be created, and the `observe`
-method of each one will be called. This allows each copy to maintain a
-state related to a single row in the batch. Remember, since each row in
-the batch is represented by a separate world, they are completely
-unrelated. This means that the model only needs to be set up in the
-original instance, and need not be shared with its copies.
-
-The `observe()` method returns a (possibly modified) version of the
-observation it sees, which are collected into a list for the agent's
-`batch_act()` method.
-
-![image](_static/img/world_batchagent.png)
-
-Now, the agent can process the entire batch at once. This is especially
-helpful for GPU-based models, which prefer to process more examples at a
-time.
+Clones are why most agents have this rough structure in their initialization
+procedures.
 
-:::{tip} Implementing batch\_act and act
```

Comment:
I do not understand how we are treating `shared` object. Do we pass `None` to the first agent, and then retrieve the `shared` attribute of that and pass to the agents? In other words, is the first agent in charge of creating the shared object, and the clones receive it in their constructor?

Reply:
Exactly. Will clarify.